### ISSUE TITLE
Bug: [@typescript-eslint/no-duplicate-enum-values] Cannot read properties of undefined

### ISSUE THREAD

**Issue by jd-carroll** [typescript-eslint/typescript-eslint#9760]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

NA

### Repro Code

```TypeScript
TSEnumDeclaration(node) {
  const enumMembers = node.body.members;
  const seenValues = new Set();
  enumMembers.forEach(member => {
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-duplicate-enum-values": ["error"],
  },
};
```

### Expected Result

The lint rule not to fail

### Actual Result

```
[Error - 10:08:04 AM] An unexpected error occurred:
[Error - 10:08:04 AM] TypeError: Cannot read properties of undefined (reading 'members')
Occurred while linting /develop/project/client-workbench/features/client-workbench-core/src/contexts/OrganizationContext/OrganizationContextProvider.tsx:31
Rule: "@typescript-eslint/no-duplicate-enum-values"
    at TSEnumDeclaration (/develop/project/client-workbench/node_modules/@project/eslint-config/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-duplicate-enum-values.js:33:47)
    at ruleErrorHandler (/develop/project/client-workbench/node_modules/eslint/lib/linter/linter.js:1076:28)
    at /develop/project/client-workbench/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/develop/project/client-workbench/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/develop/project/client-workbench/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/develop/project/client-workbench/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/develop/project/client-workbench/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/develop/project/client-workbench/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:803:23)
    at /develop/project/client-workbench/node_modules/eslint/lib/linter/linter.js:1111:32
```

### Additional Info

First, I am very suspect that this is an issue of my own making.

However, I've tried updating everything to latest (except core eslint -> we have not had the time to migrate to v9 we are on v8.57.0) and the problem persists.

The issue is that the shape of the `node` received by `TSEnumDeclaration`, looks like: 
```
{
  type: 'TSEnumDeclaration',
  const: false,
  declare: false,
  id: {
    ...
  },
  members: [
    {
      type: 'TSEnumMember',
      computed: false,
      id: [Object],
      initializer: [Object],
      range: [Array],
      loc: [Object],
      parent: [Circular *1]
    },
    ...
  ],
  range: [ 976, 1181 ],
  loc: { start: { line: 31, column: 7 }, end: { line: 36, column: 1 } },
  parent: {
    ...
  }
}
```

ie. There is no `body`, so obviously this would break.

So I guess the question is, **did something change in eslint v8 that is causing this?** If not, **did something else change and this is an actual bug in the rule?**

**Comment by jd-carroll** [comment]
Sorry, I've updated the versions above (I miss-state our storybook versions which we have also not updated 🙃)

We are on eslint v8.57.0

**Comment by JoshuaKGoldberg** [comment]
> Playground Link
> NA
>
> Repro Code
> (internal source code from typescript-eslint)

By "repro" we're looking for _your_ code that is causing this error 😄. Otherwise we can't really reproduce it. Request: please send us a way to reproduce this crash, such as a playground link.

**Comment by jd-carroll** [comment]
I get that, I will try to come up with a repro but I'm a solo dev working on a project (not funded by anyone) so I'm more so reporting to maybe help the community.

Here is the patch I ended up putting in place to resolve the issue:
```patch
diff --git a/dist/rules/no-duplicate-enum-values.js b/dist/rules/no-duplicate-enum-values.js
index 026bd4db2e3d1c71436e06ea74d38aa03fc5fd4d..5f4d179c6dae74c9e20b927ecf613c25047ffdfa 100644
--- a/dist/rules/no-duplicate-enum-values.js
+++ b/dist/rules/no-duplicate-enum-values.js
@@ -26,7 +26,7 @@ exports.default = (0, util_1.createRule)({
         }
         return {
             TSEnumDeclaration(node) {
-                const enumMembers = node.body.members;
+                const enumMembers = node.body?.members ?? node.members;
                 const seenValues = new Set();
                 enumMembers.forEach(member => {
                     if (member.initializer === undefined) {
diff --git a/dist/rules/no-mixed-enums.js b/dist/rules/no-mixed-enums.js
index 041a4ad47998df3634777c7c8545580fcd3be564..fa346046f361eebea09186828d456626398966f1 100644
--- a/dist/rules/no-mixed-enums.js
+++ b/dist/rules/no-mixed-enums.js
@@ -60,9 +60,9 @@ exports.default = (0, util_1.createRule)({
             };
             let scope = context.sourceCode.getScope(node);
             for (const definition of scope.upper?.set.get(name)?.defs ?? []) {
+                const members = definition.node.body?.members ?? definition.node.members;
                 if (definition.node.type === utils_1.AST_NODE_TYPES.TSEnumDeclaration &&
-                    definition.node.range[0] < node.range[0] &&
-                    definition.node.body.members.length > 0) {
+                    definition.node.range[0] < node.range[0] && members.length > 0) {
                     found.previousSibling = definition.node;
                     break;
                 }
@@ -129,7 +129,8 @@ exports.default = (0, util_1.createRule)({
             // enum MyEnum { A }
             // enum MyEnum { B }
             if (previousSibling) {
-                return getMemberType(previousSibling.body.members[0]);
+                const members = previousSibling.body?.members ?? previousSibling.members;
+                return getMemberType(members[0]);
             }
             // Case: Namespace declaration merging
             // namespace MyNamespace {
@@ -157,18 +158,20 @@ exports.default = (0, util_1.createRule)({
                     : AllowedType.Number;
             }
             // Finally, we default to the type of the first enum member
-            return getMemberType(node.body.members[0]);
+            const members = node.body?.members ?? node.members;
+            return getMemberType(members[0]);
         }
         return {
             TSEnumDeclaration(node) {
-                if (!node.body.members.length) {
+                const members = node.body?.members ?? node.members;
+                if (!members.length) {
                     return;
                 }
                 let desiredType = getDesiredTypeForDefinition(node);
                 if (desiredType === ts.TypeFlags.Unknown) {
                     return;
                 }
-                for (const member of node.body.members) {
+                for (const member of members) {
                     const currentType = getMemberType(member);
                     if (currentType === AllowedType.Unknown) {
                         return;
diff --git a/dist/rules/prefer-enum-initializers.js b/dist/rules/prefer-enum-initializers.js
index 39f8b7e2c905d3b145aed39fc2cd900cb129e087..05ecb4f025e4edf51c3fd955ce263fc97ba1d579 100644
--- a/dist/rules/prefer-enum-initializers.js
+++ b/dist/rules/prefer-enum-initializers.js
@@ -18,7 +18,7 @@ exports.default = (0, util_1.createRule)({
     defaultOptions: [],
     create(context) {
         function TSEnumDeclaration(node) {
-            const { members } = node.body;
+            const { members } = node.body ?? node;
             members.forEach((member, index) => {
                 if (member.initializer == null) {
                     const name = context.sourceCode.getText(member);
diff --git a/dist/rules/prefer-literal-enum-member.js b/dist/rules/prefer-literal-enum-member.js
index 51c096d54f6f1f6c2b3da4b663dfee2a7e78e1ce..1f273b23fe0a18bdcbfc229c687d5bb870633c0e 100644
--- a/dist/rules/prefer-literal-enum-member.js
+++ b/dist/rules/prefer-literal-enum-member.js
@@ -36,7 +36,8 @@ exports.default = (0, util_1.createRule)({
             return node.type === utils_1.AST_NODE_TYPES.Identifier && node.name === name;
         }
         function hasEnumMember(decl, name) {
-            return decl.body.members.some(member => isIdentifierWithName(member.id, name) ||
+            const members = decl.body?.members ?? decl.members;
+            return members.some(member => isIdentifierWithName(member.id, name) ||
                 (member.id.type === utils_1.AST_NODE_TYPES.Literal &&
                     (0, util_1.getStaticStringValue)(member.id) === name));
         }

```

**Comment by bradzacher** [comment]
My guess: you've got v7 of our parser installed and v8 of the plugin. This state is broken and not supported. You must keep all packages in our project at the exact same version.
